embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/session/orchestrator/SessionOrchestratorImpl.kt

```diff
@@ -1,19 +1,36 @@
 package io.embrace.android.embracesdk.session.orchestrator
 
+import io.embrace.android.embracesdk.config.ConfigService
+import io.embrace.android.embracesdk.internal.MessageType
 import io.embrace.android.embracesdk.internal.clock.Clock
 import io.embrace.android.embracesdk.session.BackgroundActivityService
+import io.embrace.android.embracesdk.session.ConfigGate
 import io.embrace.android.embracesdk.session.SessionService
 import io.embrace.android.embracesdk.session.lifecycle.ProcessStateService
 
 internal class SessionOrchestratorImpl(
     private val processStateService: ProcessStateService,
-    private val sessionService: SessionService,
-    private val backgroundActivityService: BackgroundActivityService?,
-    clock: Clock
+    sessionServiceImpl: SessionService,
+    backgroundActivityServiceImpl: BackgroundActivityService?,
+    clock: Clock,
+    configService: ConfigService
 ) : SessionOrchestrator {
 
+    private val sessionGate = ConfigGate(sessionServiceImpl) {
+        configService.dataCaptureEventBehavior.isMessageTypeEnabled(MessageType.SESSION)
```

Comment:
The SDK can be started with sessions turned off? Is this something we actually support?